### ISSUE TITLE
gcp-org cloud build failing

### ISSUE THREAD

**Issue by dgulli** [terraform-google-modules/terraform-example-foundation#619]
### TL;DR

During 1-org step of building out the foundations, running the gcp-org git push plan (step 16 of https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/1-org), building with cloud build, the plan fails with

"ERROR: could not upload tmp_plan/*.tfplan to gs://prj-cloudbuild-artifacts-xxxx/terraform/cloudbuild/plan/xxxx; err = gsutil exited with non-zero status: 1"

### Expected behavior

expected there to be a valid tmp_plan/*.tfplan to upload and the remainder of the build to be successful

### Observed behavior

noticed this in the build log for the plan...

tmp_plan/*.tfplan: Uploading path....
CommandException: No URLs matched: tmp_plan/*.tfplan
CommandException: 1 file/object could not be transferred.
ERROR
ERROR: could not upload tmp_plan/*.tfplan

Steps 0 and 1 actually pass as green, its the closing steps of the build that fail

### Terraform Configuration

```hcl
following steps 1-17 of https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/1-org
```

### Terraform Version

```sh
have tried whats default in the repo, 0.13.6, but also upgraded to 1.1.3 to see if that was the issue. no change, reverted.
```

**Comment by bharathkkb** [comment]
Hi @dgulli 
Is the plan file generated correctly? Could you post the full cloud build logs?

**Comment by mvCap** [comment]
Hi, since I also have that problem maybe my CloudBuild output can help to solve this:
starting build "50d93463-2922-4626-b49e-632f07319e00"

FETCHSOURCE
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /workspace/.git/
From https://source.developers.google.com/p/prj-diy-terraform/r/landing-zone
 * branch            6e62d7c52597f23311441fa5c59284d9504b2dc8 -> FETCH_HEAD
HEAD is now at 6e62d7c added folder
BUILD
Starting Step #0 - "setup"
Step #0 - "setup": Pulling image: europe-west4-docker.pkg.dev/********/prj-tf-runners/terraform
Step #0 - "setup": Using default tag: latest
Step #0 - "setup": latest: Pulling from *********/prj-tf-runners/terraform
Step #0 - "setup": 2be9afee5689: Pulling fs layer
Step #0 - "setup": d8a923ffbbec: Pulling fs layer
Step #0 - "setup": 3c2cba919283: Pulling fs layer
Step #0 - "setup": 23132c3701e2: Pulling fs layer
Step #0 - "setup": 7bcdb01842b8: Pulling fs layer
Step #0 - "setup": 56f3b93c905b: Pulling fs layer
Step #0 - "setup": a27ef3f3c972: Pulling fs layer
Step #0 - "setup": 0df797df0710: Pulling fs layer
Step #0 - "setup": 23132c3701e2: Waiting
Step #0 - "setup": 7bcdb01842b8: Waiting
Step #0 - "setup": 0df797df0710: Waiting
Step #0 - "setup": 56f3b93c905b: Waiting
Step #0 - "setup": a27ef3f3c972: Waiting
Step #0 - "setup": 3c2cba919283: Verifying Checksum
Step #0 - "setup": 3c2cba919283: Download complete
Step #0 - "setup": 2be9afee5689: Verifying Checksum
Step #0 - "setup": 2be9afee5689: Download complete
Step #0 - "setup": 2be9afee5689: Pull complete
Step #0 - "setup": d8a923ffbbec: Verifying Checksum
Step #0 - "setup": d8a923ffbbec: Download complete
Step #0 - "setup": d8a923ffbbec: Pull complete
Step #0 - "setup": 3c2cba919283: Pull complete
Step #0 - "setup": 7bcdb01842b8: Verifying Checksum
Step #0 - "setup": 7bcdb01842b8: Download complete
Step #0 - "setup": 56f3b93c905b: Verifying Checksum
Step #0 - "setup": 56f3b93c905b: Download complete
Step #0 - "setup": 0df797df0710: Verifying Checksum
Step #0 - "setup": 0df797df0710: Download complete
Step #0 - "setup": 23132c3701e2: Verifying Checksum
Step #0 - "setup": 23132c3701e2: Download complete
Step #0 - "setup": a27ef3f3c972: Download complete
Step #0 - "setup": 23132c3701e2: Pull complete
Step #0 - "setup": 7bcdb01842b8: Pull complete
Step #0 - "setup": 56f3b93c905b: Pull complete
Step #0 - "setup": a27ef3f3c972: Pull complete
Step #0 - "setup": 0df797df0710: Pull complete
Step #0 - "setup": Digest: sha256:f1f1430683400af81d2d0760fef562cf9e908b4d48a71436189a3303d21cacde
Step #0 - "setup": Status: Downloaded newer image for europe-west4-docker.pkg.dev/*********/prj-tf-runners/terraform:latest
Step #0 - "setup": europe-west4-docker.pkg.dev/*********/prj-tf-runners/terraform:latest
Step #0 - "setup": Setting up gcloud for impersonation
Step #0 - "setup": Updated property [auth/impersonate_service_account].
Step #0 - "setup": Adding bucket information to backends
Finished Step #0 - "setup"
Starting Step #1 - "tf plan validate all"
Step #1 - "tf plan validate all": Already have image (with digest): europe-west4-docker.pkg.dev/*********/prj-tf-runners/terraform
Step #1 - "tf plan validate all": .git/branches doesn't match ^(development|non-production|production|shared)$; skipping
Step #1 - "tf plan validate all": .git/refs doesn't match ^(development|non-production|production|shared)$; skipping
Step #1 - "tf plan validate all": .git/objects doesn't match ^(development|non-production|production|shared)$; skipping
Step #1 - "tf plan validate all": .git/hooks doesn't match ^(development|non-production|production|shared)$; skipping
Step #1 - "tf plan validate all": .git/info doesn't match ^(development|non-production|production|shared)$; skipping
Step #1 - "tf plan validate all": .git/logs doesn't match ^(development|non-production|production|shared)$; skipping
Finished Step #1 - "tf plan validate all"
PUSH
Artifacts will be uploaded to gs://prj-cloudbuild-artifacts-********* using gsutil cp
cloudbuild-tf-plan.yaml: Uploading path....
Copying file://cloudbuild-tf-plan.yaml [Content-Type=application/octet-stream]...
/ [0/1 files][    0.0 B/  1.7 KiB]   0% Done                                    
/ [1/1 files][  1.7 KiB/  1.7 KiB] 100% Done                                    
Operation completed over 1 objects/1.7 KiB.                                      
cloudbuild-tf-plan.yaml: 1 matching files uploaded
tmp_plan/*.tfplan: Uploading path....
CommandException: No URLs matched: tmp_plan/*.tfplan
CommandException: 1 file/object could not be transferred.
ERROR
ERROR: could not upload tmp_plan/*.tfplan to gs://prj-cloudbuild-artifacts-*********/terraform/cloudbuild/plan/*********0/; err = gsutil exited with non-zero status: 1

**Comment by mvCap** [comment]
So for me the error was because there was no "envs/" folder that contained either development/production/shared services. Once I added a folder the tfwrapper script worked and the "tmp_plan" folder got created.

**Comment by organic-data** [comment]
Had the same issue, when I'd made a mistake in copying the 1-org also in the repo, which do not belong there.

**Comment by dgulli** [comment]
hey @organic-data could you elaborate a little more about the mistake if thats ok?

**Comment by organic-data** [comment]
@dgulli Sure, the tf_wrapper.sh script searches for the 'shared' directory in the 'envs' directory in the gcp-org repository, but I'd copied them with the parent folder 1-org by mistake, did miss the last slash, and this caused the issue above.
After fixing the directory structure I was able to run Cloud Build.

**Comment by dgulli** [comment]
Amazing, thanks - will try this out.
